sdks/java/maven-archetypes/examples-java8/src/main/resources/archetype-resources/pom.xml

```diff
@@ -50,7 +49,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.5.1</version>
+        <version>@maven-compiler-plugin.version@</version>
```

Comment:
This is fine, for sure. But did you consider adding them as properties up top? The silly looking pattern of `<surefire-plugin.version>@surefire-plugin.version</surefire-plugin.version>`. The layer of indirection seems to be a pretty useful practice for keeping versions close to each other and easy to read.